File: src/main/java/com/github/unldenis/hologram/IHologramPool.java
package com.github.unldenis.hologram;

import java.util.Collection;
import org.bukkit.plugin.Plugin;
import org.jetbrains.annotations.NotNull;

public interface IHologramPool {

  void takeCareOf(Hologram hologram);

  Plugin getPlugin();

  boolean remove(@NotNull Hologram hologram);

  Collection<Hologram> getHolograms();

}

File: src/main/java/com/github/unldenis/hologram/IHologramPool.kt
package com.github.unldenis.hologram

import org.bukkit.plugin.Plugin

interface IHologramPool {

    val plugin: Plugin
    val holograms: Collection<Hologram>

    fun takeCareOf(hologram: Hologram)

    fun remove(hologram: Hologram): Boolean


}
